src/main/java/shef/data/Recipe.java

```diff
@@ -209,6 +222,33 @@ protected boolean isValidStepPosition(int position) {
     return position >= 0 && position < steps.size();
   }
 
+  /** Returns the tags of an EmbeddedEntity as a Set. */
+  private Set<String> getTagsFromEntity(Collection<EmbeddedEntity> entityTags) {
+    Set<String> tagsSet = new HashSet<>();
+    for (EmbeddedEntity tag : entityTags) {
+      tagsSet.add((String) tag.getProperty("tag"));
+    }
+    return tagsSet;
+  }
+
+  /** Returns the ingredients of an EmbeddedEntity as a Set. */
+  private Set<String> getIngredientsFromEntity(Collection<EmbeddedEntity> entityIngredients) {
+    Set<String> ingredientsSet = new HashSet<>();
+    for (EmbeddedEntity ingredient : entityIngredients) {
+      ingredientsSet.add((String) ingredient.getProperty("ingredient"));
+    }
+    return ingredientsSet;
+  }
+
+  /** Returns the steps of an EmbeddedEntity as a List. */
```

Comment:
Is there a special reason this is a list instead of a set, as for ingredients and tags?

Reply:
If Steps are saved in a Set, their order isn't preserved, which causes an original recipe's steps to appear out of order when a user tries to create a spin-off.